### PR TITLE
Fixes several Shaded Hills map issues

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Fixes several issues from #4191.
- Renames several daycycle vars/types to be clearer.
- Increases length of default daycycle to an hour, and shaded hills to 2 hours. The effective daylight period is less than half the total, so it was very short on Shadyhills.
- Uncommented weather transition weighting and tweaked the weightings.
- Allowed weather to be supplied a list of states to forbid, and forbids snow on Shaded Hills.
- Moves Shaded Hills weather into a new finalize_map_generation() proc on /datum/map, called at the end of SSmapping Initialize().
- Adds GLUT_TINY to drakes so they can eat mice. This is important apparently.

### CODE REVIEW

File: code/modules/weather/weather_fsm.dm [hunk 1]
```diff
@@ -6,5 +6,5 @@
 	for(var/decl/state_transition/weather/state_transition in valid_transitions)
 		transitions[state_transition] = state_transition.likelihood_weighting
 	if(length(transitions))
-		return pick(transitions) //pickweight(transitions)
+		return pickweight(transitions)
```
Comment (out-of-phaze) [inline code/modules/weather/weather_fsm.dm:9]: ***WHAT***

**Review by out-of-phaze — APPROVED** [review]

File: code/modules/weather/weather_fsm.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/modules/weather/weather_fsm.dm:9]: What?

Comment (out-of-phaze) [reply code/modules/weather/weather_fsm.dm:9]: i had no clue the transition weighting was disabled...

Comment (MistakeNot4892) [reply code/modules/weather/weather_fsm.dm:9]: I thought pickweight was broken.

**Pull request merged**